src/awkward/contents/emptyarray.py

```diff
@@ -85,11 +85,13 @@ def _repr(self, indent, pre, post):
     def to_NumpyArray(self, dtype, backend=None):
         backend = backend or self._backend
         return ak.contents.NumpyArray(
-            backend.nplike.empty(0, dtype), parameters=self._parameters, backend=backend
+            backend.nplike.empty(0, dtype=dtype),
+            parameters=self._parameters,
+            backend=backend,
         )
 
     def __array__(self, **kwargs):
-        return numpy.empty((0,))
+        return numpy.empty(0, dtype=np.float64)
```

Comment:
@jpivarski this is correct, yes?

Reply:
This is correct. The reason I was left the dtype implicit (which is, in practice, `np.float64`) is because when Awkward's `UnknownType` has to become a concrete type, we want it to become whatever NumPy's default is when a dtype is not specified.

However, we _know_ that NumPy's default is `np.float64`, and I think that fact is used elsewhere in the code (`to_NumpyArray`?). NumPy can't change their default, so I think it's fine to hard-code it like this, this that's more informative to readers of this codebase.